src/mlpack/methods/edge_boxes/feature_extraction_impl.hpp

```diff
@@ -0,0 +1,916 @@
+/**
+ * @file feature_extraction_Impl.hpp
+ * @author Nilay Jain
+ *
+ * Implementation of feature extraction methods.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_IMPL_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_IMPL_HPP
+
+
+#include "feature_extraction.hpp"
+#include <mlpack/methods/pca/pca.hpp>
+
+namespace mlpack {
+namespace structured_tree {
+
+
+template<typename MatType, typename CubeType>
+StructuredForests<MatType, CubeType>::
+StructuredForests(mlpack::structured_tree::FeatureParameters F) : params(F) {}
+
+
+template<typename MatType, typename CubeType>
+void StructuredForests<MatType, CubeType>::
+GetFeatureDimension(arma::vec& FtrDim)
+{
+  FtrDim = arma::vec(2);
+
+  const size_t shrink = this->params.Shrink();
+  const size_t pSize = this->params.PSize();
+  const size_t numCell = this->params.NumCell();
+  const size_t numOrient = this->params.NumOrient();
+  const size_t nColorCh = params.RGBD() == 0 ? 3 : 4;
+  const size_t nCh = nColorCh + 2 * (1 + numOrient);
+  FtrDim[0] = std::pow((pSize / shrink) , 2) * nCh;
+  FtrDim[1] = std::pow(numCell , 2) * (std::pow (numCell, 2) - 1) / 2 * nCh;
+}
+
+
+template<typename MatType, typename CubeType>
+void StructuredForests<MatType, CubeType>::
+DistanceTransform1D(const arma::vec& f, const size_t n, const double inf,
```

Comment:
Is this function from the "Distance Transforms of Sampled Functions" paper? If that's the case, we should cite the paper, as we did here: https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/dropout_layer.hpp. Also do you think, we could define inf inside the function and probably use DBL_MAX and -DBL_MAX instead of 999999.99?